### PR TITLE
Fix for concurrency bug

### PR DESCRIPTION
**Pull request by FeFelten**
#159

### CODE REVIEW

**Comment by thellmund** [comment]
Hi there ✌️ Thanks for taking the time to fix this issue, but with a recent change that I’ve made in `master`, I believe that this will no longer be an issue as soon as I release the next version.

**Comment by FeFelten** [comment]
Hi, with your current master branch I still can reproduce the bug. So the concurrency issue seems not to be solved.

**Comment by FeFelten** [comment]
I resolved the merge conflicts.

File: core/src/main/java/com/alamkanak/weekview/EventChipsCache.kt [hunk 1]
```diff
@@ -2,14 +2,15 @@ package com.alamkanak.weekview
...
+import java.util.concurrent.LinkedBlockingQueue
 
 internal class EventChipsCache {
 
     val allEventChips: List<EventChip>
         get() = normalEventChipsByDate.values.flatten() + allDayEventChipsByDate.values.flatten()
 
-    private val normalEventChipsByDate = ConcurrentHashMap<Long, MutableList<EventChip>>()
-    private val allDayEventChipsByDate = ConcurrentHashMap<Long, MutableList<EventChip>>()
+    private val normalEventChipsByDate = ConcurrentHashMap<Long, LinkedBlockingQueue<EventChip>>()
```
Comment (thellmund) [inline core/src/main/java/com/alamkanak/weekview/EventChipsCache.kt:12]: I’m a bit confused by the usage of `LinkedBlockingQueue` here, as we’re not using any queue properties or operations. Have you considered alternatives?

Comment (FeFelten) [reply core/src/main/java/com/alamkanak/weekview/EventChipsCache.kt:12]: Basically it is not necessary to use a Queue but I didn't find a standard implementation of a concurrency-aware list. That's why I used this Queue.

**Comment by thellmund** [comment]
I ran the code that you provided in the issue on master and couldn't see a crash. Can you share the API level for which it happens?

**Comment by FeFelten** [comment]
I tested it on a Nexus 10 Emulator with API 28. 

Just start the app, then you will see some flickering of the fast changing events. Afterwards scroll around in all four directions and after a few seconds the app crashes, with the described error.

I pushed my branch to reproduce the bug to https://github.com/FeFelten/Android-Week-View/tree/reproduce_concurrency_bug.

**Comment by thellmund** [comment]
Alright, am able to reproduce it now. I’m still a bit unhappy with using a `Queue`. `CopyOnWriteArrayList` might be an alternative. What do you think?

**Comment by FeFelten** [comment]
Yes, I tested it and it works with CopyOnWriteArrayList. I updated the Pull Request and minimized the changes.

**Comment by thellmund** [comment]
Nice, then let’s go ahead with it. Thanks for contributing!

**Pull request merged**